relentless/potential/pair.py

```diff
@@ -561,3 +561,244 @@ def _derivative(self, param, r, epsilon, kappa, **params):
+            The P parameter for the potential function.
+        sigma_i : float or int
+            The sigma_i parameter for the potential function.
+        sigma_j : float or int
+            The sigma_j parameter for the potential function.
+        sigma_d : float or int
+            The sigma_d parameter for the potential function.
+
+        Returns
+        -------
+        array_like
+            Returns the value or values of the energy evaluated at r.
+
+        Raises
+        ------
+        ValueError
+            If sigma_i, sigma_j, and sigma_d are not all positive.
+
+        """
+        if sigma_i<=0 or sigma_j<=0 or sigma_d<=0:
+            raise ValueError('sigma_i, sigma_j, and sigma_d must all be positive')
+        r,u,s = self._zeros(r)
+
+        #clamp lo
+        lo = 0.5*(sigma_i + sigma_j)
+        p1 = (0.5*(sigma_i + sigma_j) + sigma_d - lo)**2
+        p2 = lo**2 + lo*(sigma_i + sigma_j + 2*sigma_d) - 0.75*(sigma_i - sigma_j)**2
+        u_lo = -(np.pi*P*p1*p2)/(12*lo)
+        f_lo  = self._force(r=lo, P=P, sigma_i=sigma_i, sigma_j=sigma_j, sigma_d=sigma_d)
+        u[np.less(r,lo)] = u_lo - f_lo*(r[np.less(r,lo)] - lo)
+
+        #clamp hi
+        hi = lo + sigma_d
+        u[np.greater(r,hi)] = 0
+
+        # evaluate in between
+        flags = np.logical_and(np.greater_equal(r,lo), np.less_equal(r,hi))
+        p1 = (0.5*(sigma_i + sigma_j) + sigma_d - r[flags])**2
+        p2 = r[flags]**2 + r[flags]*(sigma_i + sigma_j + 2*sigma_d) - 0.75*(sigma_i - sigma_j)**2
+        u[flags] = -(np.pi*P*p1*p2)/(12*r[flags])
+
+        if s:
+            u = u.item()
+        return u
+
+    def _force(self, r, P, sigma_i, sigma_j, sigma_d, **params):
+        """Evaluates the depletion force.
+
+        Parameters
+        ----------
+        r : array_like
+            The value or values of r at which to evaluate the force.
+        P : float or int
+            The P parameter for the potential function.
+        sigma_i : float or int
+            The sigma_i parameter for the potential function.
+        sigma_j : float or int
+            The sigma_j parameter for the potential function.
+        sigma_d : float or int
+            The sigma_d parameter for the potential function.
+
+        Returns
+        -------
+        array_like
+            Returns the value or values of the force evaluated at r.
+
+        Raises
+        ------
+        ValueError
+            If sigma_i, sigma_j, and sigma_d are not all positive.
+
+        """
+        if sigma_i<=0 or sigma_j<=0 or sigma_d<=0:
+            raise ValueError('sigma_i, sigma_j, and sigma_d must all be positive')
+        r,f,s = self._zeros(r)
+        lo = 0.5*(sigma_i + sigma_j)
+        hi = lo + sigma_d
+
+        #clamp lo
+        lo = 0.5*(sigma_i + sigma_j)
+        p1 = lo**2 - 0.25*(sigma_i - sigma_j)**2
+        p2 = (0.5*(sigma_i + sigma_j) + sigma_d)**2 - lo**2
+        f[np.less(r,lo)] = -(np.pi*P*p1*p2)/(4*lo**2)
```

Comment:
Again you could use the contact force:

`f_lo = -np.pi*P*sigma_i*sigma_j*sigma_d*(sigma_i+sigma_j+sigma_d)/(sigma_i+sigma_j)**2`

I don't know if this is "simpler" here, but as with the contact potential, this expression can help simplify `_derivative`.